Framework/Algorithms/src/CreateLogTimeCorrection.cpp

```diff
@@ -77,49 +78,41 @@ void CreateLogTimeCorrection::exec() {
 //----------------------------------------------------------------------------------------------
 /** Get instrument geometry setup including L2 for each detector and L1
   */
-void CreateLogTimeCorrection::getInstrumentSetup() {
-  // 1. Get sample position and source position
-  IComponent_const_sptr sample = m_instrument->getSample();
-  if (!sample) {
-    throw runtime_error("No sample has been set.");
-  }
-  V3D samplepos = sample->getPos();
-
-  IComponent_const_sptr source = m_instrument->getSource();
-  if (!source) {
-    throw runtime_error("No source has been set.");
-  }
-  V3D sourcepos = source->getPos();
-  m_L1 = sourcepos.distance(samplepos);
+double
+CreateLogTimeCorrection::getInstrumentSetup(const DetectorInfo &detectorInfo) {
```

Comment:
This function seems a bit useless now. Have you considered removing `m_l2map` and `m_correctionMap` and simply using `DetectorInfo`? That would get rid of at least 2 map lookups.